addons/website/static/src/components/website_loader/website_loader.js

```diff
@@ -1,29 +1,214 @@
 /** @odoo-module **/
 
-const { EventBus, Component, useState } = owl;
+import { useService } from "@web/core/utils/hooks";
+import { sprintf } from "@web/core/utils/strings";
+import { _t } from "@web/core/l10n/translation";
+const { EventBus, Component, markup, useEffect, useState } = owl;
 
 export class WebsiteLoader extends Component {
     setup() {
+        this.rpc = useService("rpc");
+
         const initialState = {
             isVisible: false,
             title: '',
             showTips: false,
+            selectedFeatures: [],
         };
+        let tipsInterval;
+        let trackModulesInterval;
 
         this.state = useState({
             ...initialState,
         });
+        this.waitingTips = useState([]);
+        this.featuresInstallInfo = useState({ nbInstalled: 0, total: undefined });
+
+        useEffect(
+            (selectedFeatures) => {
+                // Populate this.waitingTips with the relevant ones
+                const currentWaitingTips = selectedFeatures?.length > 0 ?
+                    this.getWaitingTips(selectedFeatures) :
+                    this.getWaitingTips([]);
+
+                this.waitingTips.splice(0, this.waitingTips.length, ...currentWaitingTips);
+
+                // Request the number of modules/dependencies to install
+                // and already installed
+                if (selectedFeatures?.length > 0) {
+                    this.trackModules(selectedFeatures).catch(console.error);
+                    trackModulesInterval = setInterval(() => {
+                        this.trackModules(selectedFeatures).catch(console.error);
+                        if (this.featuresInstallInfo.nbInstalled === this.featuresInstallInfo.total) {
+                            clearInterval(trackModulesInterval);
+                        }
+                    }, 2000);
+                }
+
+                return () => clearInterval(trackModulesInterval);
+            },
+            () => [this.state.selectedFeatures]
+        );
+
+        // Cycle through the waitingTips every 10s
+        useEffect(
+            () => {
+                if (this.waitingTips.length > 0) {
+                    tipsInterval = setInterval(() => {
+                        const currentTip = this.waitingTips.find(tip => tip.show);
+                        const nextTip =
+                            this.waitingTips[this.waitingTips.indexOf(currentTip) + 1]
+                            ?? this.waitingTips[0];
```

Comment:
About the looping: I asked PDE about it already, his answer was "nevermind". It's not absolutely ideal, but not a big deal either.
About trying to match the ratio of installed features with the tip displayed: since it's not a smooth curve and some features / dependencies took way less time than others, this could switch too quickly, or skip some tips, and generally speaking in terms of UX I think it's better to change on a regular basis (at least more or less) than have widely different timings. Especially because the first modules are the ones which take more time, and the idea is to tell the user "be patient, it's working", so we need to convey this message soon enough. By the time it loops, the progress bar should be advanced enough that the tips don't really matter anymore.